File: src/main/java/com/netsensia/rivalchess/config/Extensions.java
package com.netsensia.rivalchess.config;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public enum Extensions {

    FRACTIONAL_EXTENSION_FULL (8),
    FRACTIONAL_EXTENSION_THREAT (8),
    FRACTIONAL_EXTENSION_CHECK (8),
    FRACTIONAL_EXTENSION_RECAPTURE (8),
    FRACTIONAL_EXTENSION_PAWN (8),
    RECAPTURE_EXTENSION_MARGIN (50),
    LAST_EXTENSION_LAYER (4)
    ;

    private int value;

    private Extensions(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static List<Integer> getMaxNewExtensionsTreePart() {
        return Collections.unmodifiableList(Arrays.asList(
                Extensions.FRACTIONAL_EXTENSION_FULL.getValue(),
                Extensions.FRACTIONAL_EXTENSION_FULL.getValue() / 4 * 3,
                Extensions.FRACTIONAL_EXTENSION_FULL.getValue() / 2,
                Extensions.FRACTIONAL_EXTENSION_FULL.getValue() / 8, 0));
    }
}

File: src/main/java/com/netsensia/rivalchess/config/Extensions.kt
package com.netsensia.rivalchess.config

import java.util.*

const val FRACTIONAL_EXTENSION_FULL = 8
const val FRACTIONAL_EXTENSION_THREAT = 8
const val FRACTIONAL_EXTENSION_CHECK = 8
const val FRACTIONAL_EXTENSION_RECAPTURE = 8
const val FRACTIONAL_EXTENSION_PAWN = 8
const val RECAPTURE_EXTENSION_MARGIN = 50
const val LAST_EXTENSION_LAYER = 4

val maxNewExtensionsTreePart = intArrayOf(
                    FRACTIONAL_EXTENSION_FULL,
                    FRACTIONAL_EXTENSION_FULL / 4 * 3,
                    FRACTIONAL_EXTENSION_FULL / 2,
                    FRACTIONAL_EXTENSION_FULL / 8, 0)
